src/write.c

```diff
@@ -1291,19 +1430,12 @@ avifResult avifEncoderFinish(avifEncoder * encoder, avifRWData * output)
             }
         }
 
-        uint32_t imageWidth = imageMetadata->width;
-        uint32_t imageHeight = imageMetadata->height;
-        if (isGrid) {
-            imageWidth = imageMetadata->width * item->gridCols;
-            imageHeight = imageMetadata->height * item->gridRows;
-        }
-
         // Properties all av01 items need
 
         avifItemPropertyDedupStart(dedup);
         avifBoxMarker ispe = avifRWStreamWriteFullBox(&dedup->s, "ispe", AVIF_BOX_SIZE_TBD, 0, 0);
-        avifRWStreamWriteU32(&dedup->s, imageWidth);  // unsigned int(32) image_width;
-        avifRWStreamWriteU32(&dedup->s, imageHeight); // unsigned int(32) image_height;
+        avifRWStreamWriteU32(&dedup->s, item->outputWidth);  // unsigned int(32) image_width;
+        avifRWStreamWriteU32(&dedup->s, item->outputHeight); // unsigned int(32) image_height;
```

Comment:
Tracks are a new topic to me and I am not so sure of using `item->outputWidth` here instead of `imageWidth` (same at lines 1587 and 1701). There is not enough test coverage for tracks with grids to know if this works either.

Reply:
> You can't have an image sequence of grids.

Can it be a grid of tracks of image sequences? Otherwise it means `isGrid` is always false here, right?

Reply:
Here is Joe's answer:

> The only way to signal a grid in any way is using a "grid" item, which refers to other items. The only way to signal a sequence is with tracks. They don't mix at all. There's no such thing as image grid sequences.

See also https://github.com/AOMediaCodec/libavif/pull/1149.

As for your second question, I am running into a limitation of GitHub's code review tool. You asked "Otherwise it means `isGrid` is always false here, right?" I assume by "here" you're referring to the following two lines:

```
        avifRWStreamWriteU32(&dedup->s, imageWidth);  // unsigned int(32) image_width;
        avifRWStreamWriteU32(&dedup->s, imageHeight); // unsigned int(32) image_height;
```

These two lines are part of the `ispe` box, which is only used with image items, not with image sequences.

Reply:
> Here is Joe's answer

I did not realize image sequences were a completely different format at the container level (we are still writing a `meta` box, though? what if there is no item?). Thank you for forwarding Joe's explanation.

> I assume by "here" you're referring to the following two lines

Sorry, I was confused. I was talking about tracks and I thought the `ispe` box also applied to them. So the grid stuff is irrelevant at the lines 1590 and 1704, good to know.

Reply:
libavif also creates an image item when encoding an image sequence. That image item is intended for renderers that cannot playback an image sequence. So there is always an item. (For simplicity libavif just points at the first key frame as a standalone image.)

At Cyril's suggestion, libavif writes a meta box for each track unless there is no data for it. (See the beginning of the `avifEncoderWriteTrackMetaBox()` function.)